src/legacy/ui/public/kfetch/kfetch.ts

```diff
@@ -61,10 +62,11 @@ export async function kfetch(
       });
 
       return window.fetch(fullUrl, restOptions).then(async res => {
-        const body = await getBodyAsJson(res);
+        const body = parseJson ? await getBodyAsJson(res) : null;
         if (res.ok) {
-          return body;
+          return parseJson ? body : res;
         }
+
```

Comment:
I think we'll want to move the kfetch changes to a standalone PR too. I see there is a merge conflict with master in this file currently

Reply:
I've altered the feedback here to make the change less-intensive.